src/Util/ParamsIterator.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * BigBlueButton open source conferencing system - https://www.bigbluebutton.org/.
+ *
+ * Copyright (c) 2016-2023 BigBlueButton Inc. and by respective authors (see below).
+ *
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free Software
+ * Foundation; either version 3.0 of the License, or (at your option) any later
+ * version.
+ *
+ * BigBlueButton is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with BigBlueButton; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace BigBlueButton\Util;
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @internal
+ *
+ * @coversNothing
+ */
+class ParamsIterator extends TestCase
+{
+    public function iterate($params, $url)
```

Comment:
```suggestion
    protected function assertAllRecursiveIterableElementsContainsStrings(iterable $haystack, string $needle): void
```
`protected`, when becoming a trait. Naming to match semantics of PHPUnit assertion methods.

Reply:
moved to TestCase, as other methods are also public, I didn't change to protected
I have chosen a different name for the method to better describe what it does